docs/key-issues/gpus.md

```diff
@@ -180,3 +204,87 @@ Potential solutions on the PyPI side include:
 - make an environment marker or selector package approach work,
 - improve interoperability with other package managers, in order to be able to
   declare a dependency on a CUDA or ROCm version as externally provided,
+
+
+## Additional Notes on CUDA Compatibility
+
+Compatibility across CUDA versions is a common problem with numerous pitfalls to be aware of.
+There are three primary components of CUDA:
+
+1. The CUDA Toolkit (CTK): This component includes the CUDA runtime library
+   (libcudart.so) along with a range of other libraries and tools including
+   math libraries like cuBLAS. libcudart.so and a few other core headers and
+   libraries are the bare minimum required to compile and link CUDA code.
+2. The User-Mode Driver (UMD): This is the libcuda.so library. This library is
+   required to load and run CUDA code.
+3. The Kernel-Mode Driver (KMD): The nvidia.ko file. This constitutes what is
+   typically considered a "driver" in common parlance when referring to other
+   peripherals connected to a computer.
+
+CUDA developers and users usually do not need to be aware of the KMD.
+For the rest of this section, when referring to the "driver" we will always be referring to the UMD.
+
+The CUDA runtime library makes no forward or backward compatibility
+guarantees, meaning that libraries that dynamically link to the CUDA runtime
```

Comment:
(Just leaving a comment, no change is needed.) TBH it's surprising that the CUDA docs described it this way. I would think it's not carefully worded. If cudart had not followed the SemVer semantics for API/ABI compatibility across minor versions (which it actually does), there would have been all sorts of issues.

Reply:
Yeah I agree that the wording is surprising and that there's definitely pretty strong compatibility. I think it's more that the CUDA team reserves the right to break this if needed, and historically they may have taken more advantage of that, but nowadays they have too many people relying on ABI compatibility to break it in practice even though they reserve the right to do so in theory.